kyaml/resid/gvk.go

```diff
@@ -10,6 +10,8 @@ import (
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
+const deprecatedBuiltinPluginApiVersion = "builtin"
```

Comment:
I cannot reference konfig.BuiltinPluginApiGroup here due to cyclic import and could not think of a better way than defining the const here...

Reply:
`kyaml` is a distinct module, usable outside of kustomize, and versioned independently.

IMO if `resid` has no other references to `builtin`, this doesn't seem like the appropriate place to add a special case. I'd adjust usages of ParseGroupVersion in the builtin transformer.

Reply:
+1 to @iamnoah's comment, kyaml is indepedent of api and should not contain code that depends on it (the dependency goes the other way around). Instead of putting a special case in in `ParseGroupVersion`, I agree that it probably would be cleaner to special case usages of `ParseGroupVersion` where this issue is showing up